### PR TITLE
add open telemetry provider

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**
fix https://github.com/istio/istio/issues/36637

API: https://github.com/istio/api/pull/2175

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/accesslog.go [hunk 1]
```diff
@@ -482,6 +488,82 @@ func buildFileAccessLogHelper(path string, mesh *meshconfig.MeshConfig) *accessl
...
+		return nil
+	}
+	attrList := make([]*otlpcommon.KeyValue, len(attrsMap))
+	index := 0
+	for key, value := range attrsMap {
+		kv := otlpcommon.KeyValue{
+			Key:   key,
+			Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: value}},
+		}
+		attrList[index] = &kv
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/accesslog.go:561]: nit: `append` is a bit less error prone. Just make sure you update L554 to `make([]*otlpcommon.KeyValue, 0, len(attrsMap))` if you do

**Review by howardjohn — COMMENTED** [review]
Nice

File: pilot/pkg/networking/core/v1alpha3/accesslog.go (hunk 1 above)
Comment (zirain) [reply pilot/pkg/networking/core/v1alpha3/accesslog.go:561]: thanks, there's a mistake, will update all this after API is merged

**Comment by zirain** [comment]
@howardjohn I think this PR is ready for review.

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**